docs/source/_src_user_guide/state_propagation/environment_setup/architecture/custom_settings.rst

```diff
@@ -0,0 +1,121 @@
+.. _custom_models:
+
+=============
+Custom models
+=============
+
+.. toctree::
+   :titlesonly:
+   :hidden:
+   :maxdepth: 1
+
+   environment_during_propagation
+
```

Comment:
Overall I do not have many comments on this page, I really like it and adding the link to the API is really a good idea!